dot/sync/tip_syncer_test.go

```diff
@@ -0,0 +1,404 @@
+			blockStateBuilder: func(ctrl *gomock.Controller) BlockState {
+				mockBlockState := NewMockBlockState(ctrl)
+				mockBlockState.EXPECT().GetHighestFinalisedHeader().Return(&types.Header{
+					Number: 2,
+				}, nil)
+				return mockBlockState
+			},
+			res: &worker{
+				startNumber:  uintPtr(3),
+				targetNumber: uintPtr(1),
+				direction:    network.Descending,
+				err:          &workerError{},
+			},
+			want: &worker{
+				startNumber:  uintPtr(3),
+				targetNumber: uintPtr(3),
+				direction:    network.Descending,
+			},
+			err: nil,
+		},
+	}
+	for name, tt := range tests {
+		tt := tt
+		ctrl := gomock.NewController(t)
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			s := &tipSyncer{
+				blockState: tt.blockStateBuilder(ctrl),
+			}
+			got, err := s.handleWorkerResult(tt.res)
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("handleWorkerResult() got = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_tipSyncer_hasCurrentWorker(t *testing.T) {
+	t.Parallel()
+
+	type args struct {
+		w       *worker
+		workers map[uint64]*worker
+	}
+	tests := map[string]struct {
+		args args
+		want bool
+	}{
+		"worker nil": {
+			want: true,
+		},
+		"ascending, false": {
+			args: args{
+				w: &worker{
+					direction:    network.Ascending,
+					startNumber:  uintPtr(2),
+					targetNumber: uintPtr(2),
+				},
+				workers: map[uint64]*worker{
+					1: {
+						direction:    network.Ascending,
+						targetNumber: uintPtr(3),
+						startNumber:  uintPtr(3),
+					},
+				},
+			},
+			want: false,
+		},
+		"ascending, true": {
+			args: args{
+				w: &worker{
+					direction:    network.Ascending,
+					startNumber:  uintPtr(2),
+					targetNumber: uintPtr(2),
+				},
+				workers: map[uint64]*worker{
+					1: {
+						direction:    network.Ascending,
+						targetNumber: uintPtr(3),
+						startNumber:  uintPtr(1),
+					},
+				},
+			},
+			want: true,
+		},
+		"descending, false": {
+			args: args{
+				w: &worker{
+					direction:    network.Descending,
+					startNumber:  uintPtr(2),
+					targetNumber: uintPtr(2),
+				},
+				workers: map[uint64]*worker{
+					1: {
+						startNumber:  uintPtr(3),
+						targetNumber: uintPtr(3),
+						direction:    network.Descending,
+					},
+				},
+			},
+			want: false,
+		},
+		"descending, true": {
+			args: args{
+				w: &worker{
+					direction:    network.Descending,
+					startNumber:  uintPtr(2),
+					targetNumber: uintPtr(2),
+				},
+				workers: map[uint64]*worker{
+					1: {
+						startNumber:  uintPtr(3),
+						targetNumber: uintPtr(1),
+						direction:    network.Descending,
+					},
+				},
+			},
+			want: true,
+		},
+	}
+	for name, tt := range tests {
+		tt := tt
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			ti := &tipSyncer{}
+			got := ti.hasCurrentWorker(tt.args.w, tt.args.workers)
```

Comment:
nit just to be consistent with other tests:

```suggestion
			s := &tipSyncer{}
			got := s.hasCurrentWorker(tt.args.w, tt.args.workers)
```